index.js

```diff
@@ -1,284 +1,161 @@
-// Following the TFD spec (also known as TFK), the naming convention in this
+// Following the BFE spec (formally known as TFK), the naming convention in this
 // file uses "T" to mean "Type byte", "TF" to mean "Type byte and Format byte"
 // and "D" to mean "Data bytes".
 
-const { isFeedType, isMsgType, isBlobType } = require('ssb-ref')
-const { definitionsToDict } = require('./util')
-const TYPES = require('./bfe.json')
-
-const NAMED_TYPES = definitionsToDict(TYPES)
-
-const FEED = NAMED_TYPES['feed']
-const FEED_T = Buffer.from([FEED.code])
-const CLASSIC_FEED_TF = Buffer.from([
-  FEED.code,
-  FEED.formats['ssb/classic'].code,
-])
-const GABBYGR_FEED_TF = Buffer.from([
-  FEED.code,
-  FEED.formats['ssb/gabby-grove'].code,
-])
-const BENDYBT_FEED_TF = Buffer.from([
-  FEED.code,
-  FEED.formats['ssb/bendy-butt'].code,
-])
-
-const MSG = NAMED_TYPES['msg']
-const MSG_T = Buffer.from([MSG.code])
-const CLASSIC_MSG_TF = Buffer.from([MSG.code, MSG.formats['ssb/classic'].code])
-const GABBYGR_MSG_TF = Buffer.from([
-  MSG.code,
-  MSG.formats['ssb/gabby-grove'].code,
-])
-const BENDYBT_MSG_TF = Buffer.from([
-  MSG.code,
-  MSG.formats['ssb/bendy-butt'].code,
-])
-
-const BLOB = NAMED_TYPES['blob']
-const BLOB_T = Buffer.from([BLOB.code])
-const CLASSIC_BLOB_TF = Buffer.from([
-  BLOB.code,
-  BLOB.formats['ssb/classic'].code,
-])
-
-const SIGNATURE = NAMED_TYPES['signature']
-const SIGNATURE_T = Buffer.from([SIGNATURE.code])
-const SIGNATURE_TF = Buffer.from([
-  SIGNATURE.code,
-  SIGNATURE.formats['ed25519'].code,
-])
-
-const BOX = NAMED_TYPES['encrypted']
-const BOX_T = Buffer.from([BOX.code])
-const BOX1_TF = Buffer.from([BOX.code, BOX.formats['box1'].code])
-const BOX2_TF = Buffer.from([BOX.code, BOX.formats['box2'].code])
-
-const GENERIC = NAMED_TYPES['generic']
-const STRING_TF = Buffer.from([
-  GENERIC.code,
-  GENERIC.formats['UTF8 string'].code,
-])
-const BOOL_TF = Buffer.from([GENERIC.code, GENERIC.formats['boolean'].code])
-const BOOL_TRUE = Buffer.from([1])
-const BOOL_FALSE = Buffer.from([0])
-const NIL_TF = Buffer.from([GENERIC.code, GENERIC.formats['nil'].code])
-const NIL_TFD = NIL_TF
+const definitions = require('./bfe.json')
+const {
+  decorateBFE,
+  definitionsToDict,
+  findClassicTypeFormat,
+} = require('./util')
+
+const TYPES = decorateBFE(definitions)
+const NAMED_TYPES = definitionsToDict(definitions)
+
+const toTF = (type, format) =>
+  Buffer.from([NAMED_TYPES[type].code, NAMED_TYPES[type].formats[format].code])
+
+const STRING_TF = toTF('generic', 'UTF8 string')
+const NIL_TF = toTF('generic', 'nil')
+const BOOL = {
+  TF: toTF('generic', 'boolean'),
+  TRUE: Buffer.from([1]),
+  FALSE: Buffer.from([0]),
+}
```

Comment:
Can you bring back BOOL_TF and BOOL_TRUE? Would be consistent with what we have already (e.g. STRING_TF) and this seems unrelated to the core of this PR

Reply:
I don't like it as much but sure